### ISSUE TITLE
Organization of statusline.lua

### ISSUE THREAD

**Issue by anott03** [wbthomason/lsp-status.nvim#40]
### The Issue
Given that the changes made in #39, the example statusline component provided in `statusline.lua` is redundant since it in many ways is rewriting those functions (though granted it has a bit more logic).

### My Proposal
What I propose is either a new module `examples.lua`, or simply adding the example to `lsp-status.lua` thus allowing it to take advantage of the individual component functions now provided.

### Potential Objections
The logic included in the example component is such that a component is only rendered if the number of errors/warnings/messages is greater than 0. Since the component functions return strings not numbers, this logic cannot remain exactly as it was, but the effect can be recreated by checking if the third character of the component is '0'.

**Comment by wbthomason** [comment]
I'm not sure I agree with the proposal, primarily because a large number of users rely on the example statusline component out of the box (it was actually originally just an example in the README, but users requested it be added to the code), and I don't want to break their setups. I would be OK with rewriting the example component in terms of the new functions, to reduce code duplication, though - as you mention - this is slightly complicated by the new functions returning strings rather than numbers/tables. What do you think?

**Comment by anott03** [comment]
I see your reasoning, though I would point out that if users are using this as intended, their config looks something like this:
```lua
local lspstatus = require('lsp-status')
...
lspstatus.status()
```
That should mean that assuming `lsp-status.lua` continues to return a function called `status`, we can refactor internally without concern.

**Comment by wbthomason** [comment]
Yep - that's more or less what I'm proposing. I'm just not in favor of removing the `status` function entirely.

**Comment by anott03** [comment]
Oh yeah. Sorry I should have been clearer about that - I don't want to remove the function entirely, just potentially out of `statusline.lua`.

**Comment by anott03** [comment]
(I didn't mean to close it)

**Comment by anott03** [comment]
> this is slightly complicated by the new functions returning strings rather than numbers/tables.

I've been thinking about this... maybe the new functions should return a table with a string and a number? I don't think it's a huge deal since with just a string it's a fairly simple matter of using `string.sub` but returning both values might be more convenient for this. However, that may make the UX a bit worse, since users would no longer be able to simply do `require('lsp-status').status_X()`.
